### PR TITLE
feat(@wallet): multi tx approval/nonce

### LINKED ISSUE #8806: Approve token when doing tx with bridge

### PR DESCRIPTION
**Pull request by alaibe**
approve transaction management

fixes #8806 
status-go https://github.com/status-im/status-go/pull/3050

### CODE REVIEW

File: src/app_service/service/transaction/service.nim [hunk 1]
```diff
@@ -304,6 +306,8 @@ QtObject:
         toAddress = parseAddress(to_addr)
       else:
         let token = self.tokenService.findTokenBySymbol(network, tokenSymbol)
+        echo token.symbol
+        echo value
```
Comment (jrainville) [inline src/app_service/service/transaction/service.nim:310]: leftover?

File: src/app_service/service/transaction/service.nim [hunk 2]
```diff
@@ -322,6 +326,19 @@ QtObject:
...
           gasFees = $route.gasFees.gasPrice
 
+        if route.approvalRequired:
+          let approve = Approve(
+            to: parseAddress(route.approvalContractAddress),
+            value: route.approvalAmountRequired,
+          )
+          data = ERC20_procS.toTable["approve"].encodeAbi(approve)
+          txData = ens_utils.buildTokenTransaction(parseAddress(from_addr), toAddress,
+  $route.gasAmount, gasFees, route.gasFees.eip1559Enabled, $route.gasFees.maxPriorityFeePerGas, $route.gasFees.maxFeePerGasM)
```
Comment (jrainville) [inline src/app_service/service/transaction/service.nim:336]: indentation looks weird here

**Review by jrainville — COMMENTED** [review]
Looks good. 2 small comments

**Review by Khushboo-dev-cpp — COMMENTED** [review]
Looks good! I think we dont need this new check image you have added though?

https://github.com/status-im/status-desktop/blob/master/ui/StatusQ/src/assets/img/icons/tiny/tiny-checkmark.svg

does this one work instead as it is already present?

File: src/app_service/service/transaction/service.nim (hunk 1 above)
Comment (alaibe) [reply src/app_service/service/transaction/service.nim]: updated

File: src/app_service/service/transaction/service.nim (hunk 2 above)
Comment (alaibe) [reply src/app_service/service/transaction/service.nim]: yes indeed, updated

**Comment by alaibe** [comment]
@Khushboo-dev-cpp I did reuse the existing one

**Comment by anastasiyaig** [comment]
@richard-ramos your kind review is appreciated here :)

**Review by jrainville — APPROVED** [review]

**Comment by anastasiyaig** [comment]
it works to me finally, however the transactions on test net are taking really loooong. But anyway , it was a success

**Review by anastasiyaig — APPROVED** [review]

**Review by richard-ramos — APPROVED** [review]
:medal_military:

**Pull request merged**